main.js

```diff
@@ -6,3 +6,64 @@ JS:
 4. bonus: clicking on the translucent background closes the dialog
 5. bonus: use Class
 */
+function dialogBox($component) {
+  const questionContainer = $component.find('.question');
+  const questionContainerHtml = questionContainer.html();
+  const socialItem = $('.social-link');
+  const buttonOk = $component.find('.button--ok');
+  const buttonCancel = $component.find('.button--cancel');
+  const dialogContainer = $('.dialog');
+  let socialLink = socialItem.attr('data-url');
+
+  function init() {
+    attachDialogBox();
+    closeDialog();
+  }
+
+  init();
+
+  function attachDialogBox() {
+    socialItem.on('click', function() {
+
+      const socialText = $(this).text();
+      socialLink = $(this).attr('data-url');
+
+      questionContainer.html(questionContainerHtml.replace('{{socialNetworkName}}', socialText));
+      questionContainer.html(questionContainer.html().replace('{{socialNetworkName}}', socialText));
+      dialogContainer.removeClass('is-hidden');
+
+      buttonOk.on('click', function() {
+
+        window.open(`https://${socialLink}`);
+
+      });
+
+      buttonCancel.on('click', function() {
+
+        dialogContainer.addClass('is-hidden');
+
+      });
+
+    });
+  }
+
+  function closeDialog() {
+    $(document).on('click', function(event) {
```

Comment:
Kudos for handling that, but the name of the function doesn't tell me it's about binding an event. I would expect that "cancel" button triggers `closeDialog` function.